master/internal/portregistry/port_registry.go

```diff
@@ -15,6 +17,13 @@ var (
 func InitPortRegistry() {
```

Comment:
can you just make this take `ReservedPorts` and pass it through?

this is pretty susceptible to someone messing with init order and implicitly breaking this